### PR TITLE
issue #701 - adding Kotlin companion object issue to api doc.

### PR DESCRIPTION
**Pull request by 10xtechie**
@mpkorstanje , @aurelien-reeves 
### 🤔 What's changed?

Update the api documentation to add the issue related to Kotlin companion objects and its solution. The updated api documentation will look like the following screenshot. 

updatedss

### ⚡️ What's your motivation? 
Keep the documentations so the other can be benefitted.

### 📋 Checklist:

- [x] I agree to respect and uphold the [Cucumber Community Code of Conduct](https://cucumber.io/conduct/)

### CODE REVIEW

**Comment by aurelien-reeves** [comment]
If you merge the main branch into yours, you should see the new warning theme instead of the danger one

**Comment by 10xtechie** [comment]
@aurelien-reeve, 
Thanks for letting me know this! I will update this pull request by merging the latest main branch into this branch. Also, I will update the description of this pull request with an updated screenshot. 
 
> If you merge the main branch into yours, you should see the new warning theme instead of the danger one

**Review by aurelien-reeves — APPROVED** [review]
It looks good for me

@mpkorstanje do you agree with the content?

File: content/docs/cucumber/api.md [hunk 1]
```diff
@@ -285,6 +285,52 @@ Before(10) { scenario: Scenario ->
...
+The @JvmStatic annotation does not prevent this behaviour of Kotlin, it adds the static methods to MyStepDefinitionsMethod only. 
+
+Consequently, Cucumber will detect the static methods in MyStepDefinitions class, as well as the non-static methodss in MyStepDefinitions$Companion class and will complain about the second one.
+
+As a soultion to this problem, [ package level functions](https://kotlinlang.org/docs/java-to-kotlin-interop.html#package-level-functions) - this is, withouth companion objects. 
+
+{{% /warn %}}
+
+```kotlin
+class MyStepDefinitions  {
```
Comment (mpkorstanje) [inline content/docs/cucumber/api.md]: This doesn't look like an example of package level methods. I assume we want to show the user how to do the right thing.

Comment (10xtechie) [reply content/docs/cucumber/api.md]: @mpkorstanje ,
Thanks for your comment! I have updated the PR accordingly, please take a look :)

File: content/docs/cucumber/api.md [hunk 2]
```diff
@@ -284,6 +284,45 @@ Before(10, () -> {
...
+```kotlin
+
+package io.cucumber.example
+
+import io.cucumber.java.AfterAll
+import io.cucumber.java.BeforeAll
+
+class MyStepDefinitions {
+    @BeforeAll
+    fun beforeAll() {
```
Comment (mpkorstanje) [inline content/docs/cucumber/api.md]: Does this actually work? To get it to work I had to put the functions at the package level. I believe this example from my comment (https://github.com/cucumber/docs/issues/701#issuecomment-983076390) worked, But I'm not sure if this one does.

Comment (10xtechie) [reply content/docs/cucumber/api.md]: @mpkorstanje ,
Thanks! changed

**Comment by mpkorstanje** [comment]
LGTM, merged with a few edits. Cheers!

**Pull request merged**